### PR TITLE
Added Firewalld Interface class

### PR DESCRIPTION
**Pull request by teclator**

### CODE REVIEW

File: library/network/src/modules/CWMFirewallInterfaces.rb [hunk 1]
```diff
@@ -250,19 +250,23 @@ def StoreAllowedInterfaces(services)
 
       zones =
         known_interfaces.each_with_object([]) do |known_interface, a|
-          if allowed_interfaces.include?(known_interface["id"])
-            zone_name = known_interface["zone"] || default_zone.name
+          puts "VAMOS CON #{known_interface.inspect}"
```
Comment (mvidner) [inline library/network/src/modules/CWMFirewallInterfaces.rb]: POR FAVOR quitar :D

Comment (teclator) [reply library/network/src/modules/CWMFirewallInterfaces.rb]: Jiaz, good leftlover. I was checking your spanish :P

File: library/network/src/lib/y2firewall/firewalld/interface.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+# ------------------------------------------------------------------------------
+
+require "yast"
+require "y2firewall/firewalld"
+
+module Y2Firewall
+  class Firewalld
+    # Class to work with firewalld interfaces
+    class Interface
+      attr_accessor :id
```
Comment (mvidner) [inline library/network/src/lib/y2firewall/firewalld/interface.rb]: Add `# @return [Symbol]`

File: library/network/src/lib/y2firewall/firewalld/interface.rb [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+      # @return [String] device name
+      def device_name
+        Yast::NetworkInterfaces.GetValue(name, "NAME")
+      end
+
+      # Return the zone name for a given interface from the firewalld instance
+      # instead of from the API.
+      #
+      # @param name [String] interface name
```
Comment (mvidner) [inline library/network/src/lib/y2firewall/firewalld/interface.rb]: param name does not exist, remove this line

File: library/network/src/lib/y2firewall/helpers/interfaces.rb [hunk 4]
```diff
@@ -44,7 +44,7 @@ def firewalld
       #
       # @return [Array<String>] default zone interface names
       def default_interfaces
-        known_interfaces.select { |i| i["zone"].to_s.empty? }.map { |i| i["id"] }
+        known_interfaces.select { |i| !i.zone }.map(&:name)
       end
 
       # Return the zone name for a given interface from the firewalld instance
```
Comment (mvidner) [inline library/network/src/lib/y2firewall/helpers/interfaces.rb]: the method `interface_zone` below could be simplified to `Y2Firewall::Firewalld::Interface.build(name).zone`

File: library/network/src/lib/y2firewall/firewalld/interface.rb [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+require "yast"
+require "y2firewall/firewalld"
+
+module Y2Firewall
+  class Firewalld
+    # Class to work with firewalld interfaces
+    class Interface
+      attr_accessor :id
+
+      def self.build(name)
```
[the remark this answers is not in the document]
Comment (teclator) [reply library/network/src/lib/y2firewall/firewalld/interface.rb]: Yep, at the end it does not make sense, so will expose the constructor again.

File: library/network/src/lib/y2firewall/helpers/interfaces.rb (hunk 4 above)
Comment (imobachgs) [reply library/network/src/lib/y2firewall/helpers/interfaces.rb]: What about using `#reject`?

```ruby
known_interfaces.reject(&:zone).map(&:name)
```

File: library/network/src/lib/y2firewall/firewalld/interface.rb [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+      # Constructor
+      #
+      # @param name [String] interface name
+      def initialize(name)
+        Yast.import "NetworkInterfaces"
+
+        @id = name.to_sym
+      end
+
+      # Return an array with all the known or configured interfaces
```
Comment (imobachgs) [inline library/network/src/lib/y2firewall/firewalld/interface.rb]: *know or configured*? I would like to have some clarification about what is exactly a "known" interface.

File: library/network/src/lib/y2firewall/helpers/interfaces.rb [hunk 7]
```diff
@@ -44,7 +44,7 @@ def firewalld
       #
       # @return [Array<String>] default zone interface names
       def default_interfaces
-        known_interfaces.select { |i| i["zone"].to_s.empty? }.map { |i| i["id"] }
+        known_interfaces.select { |i| !i.zone }.map(&:name)
```
Comment (imobachgs) [inline library/network/src/lib/y2firewall/helpers/interfaces.rb]: What about using `#reject`?

```ruby
known_interfaces.reject(&:zone).map(&:name)
```

**Pull request merged**